plugins/modules/ndo_port_channel_interface.py

```diff
@@ -0,0 +1,351 @@
+        description=dict(type="str"),
+        node=dict(type="str"),
+        interfaces=dict(type="list", elements="str", aliases=["members"]),
+        interface_policy_group=dict(
+            type="dict",
+            options=dict(
+                name=dict(type="str"),
+                template=dict(type="str"),
+            ),
+            aliases=["policy", "interface_policy"],
+        ),
+        interface_policy_group_uuid=dict(type="str", aliases=["policy_uuid", "interface_policy_uuid"]),
+        interface_descriptions=dict(
+            type="list",
+            elements="dict",
+            options=dict(
+                interface_id=dict(type="str"),
+                description=dict(type="str"),
+            ),
+        ),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["port_channel_interface", "port_channel_interface_uuid"], True],
+            ["state", "present", ["port_channel_interface", "port_channel_interface_uuid"], True],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    port_channel_interface = module.params.get("port_channel_interface")
+    port_channel_interface_uuid = module.params.get("port_channel_interface_uuid")
+    description = module.params.get("description")
+    node = module.params.get("node")
+    interfaces = module.params.get("interfaces")
+    if interfaces:
+        interfaces = ",".join(interfaces)
+    interface_policy_group = module.params.get("interface_policy_group")
+    interface_policy_group_uuid = module.params.get("interface_policy_group_uuid")
+    interface_descriptions = module.params.get("interface_descriptions")
+    state = module.params.get("state")
+
+    ops = []
+    match = None
+
+    mso_template = MSOTemplate(mso, "fabric_resource", template)
+    mso_template.validate_template("fabricResource")
+    object_description = "Port Channel Interface"
+
+    path = "/fabricResourceTemplate/template/portChannels"
+    existing_port_channel_interfaces = mso_template.template.get("fabricResourceTemplate", {}).get("template", {}).get("portChannels", [])
+    if port_channel_interface or port_channel_interface_uuid:
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            existing_port_channel_interfaces,
+            [KVPair("uuid", port_channel_interface_uuid) if port_channel_interface_uuid else KVPair("name", port_channel_interface)],
+        )
+        if match:
+            mso.existing = mso.previous = copy.deepcopy(match.details)
+    else:
+        mso.existing = mso.previous = existing_port_channel_interfaces
+
+    if state == "present":
+
+        if interface_policy_group and not interface_policy_group_uuid:
```

Comment:
if you update an existing task shouldn't it error out saying both interface_policy_group_uuid and interface_policy_group can't be present together. if not should we mention in docs, which argument will take precedence if both are present?

existing task with interface_policy_group
update task with interface_policy_group_uuid

Reply:
no it should not error out. If both are present, the UUID will be prioritized and the other one will be ignored, I will mention it in the documentation.